docs/patterns/tools/use-tools/processing-actions.mdx

```diff
@@ -127,12 +96,112 @@ agent_executor = AgentExecutor(agent=agent, tools=tools, verbose=True)
 
 agent_executor.invoke({"input": task})
 ```
-```javascript JavaScript
-Coming Soon
+</Step>
+</Steps>
+</Tab>
+<Tab title="TypeScript">
+<Steps>
+<Step title="Import required libraries">
+```typescript TypeScript
+import { ActionExecutionResDto, LangchainToolSet, RawActionData, TPostProcessor, TPreProcessor, TSchemaProcessor } from "composio-core";
+import { ChatOpenAI } from "@langchain/openai";
+import { createOpenAIFunctionsAgent, AgentExecutor } from "langchain/agents";
+import { pull } from "langchain/hub";
+import { ChatPromptTemplate } from "@langchain/core/prompts";
+```
+</Step>
+<Step title="Initialize ChatOpenAI & LangChain toolset client">
+```typescript TypeScript
+const llm = new ChatOpenAI({ apiKey: "<your-api-key>" });
+const toolset = new LangchainToolSet({ apiKey: "<your-api-key>" });
+```
+</Step>
+<Step title="Schema Modifier">
+This will be used to modify the schema of the `LINEAR_CREATE_LINEAR_ISSUE` action, we remove the parameters `project_id` and `team_id` as required fields, later in the program we will pass these values as inputs to the action manually. The technical term for this is **Schema Processing**.
+```typescript TypeScript
+const schemaProcessor: TSchemaProcessor = ({
+  actionName,
+  toolSchema,
+}: {
+  actionName: string;
+  toolSchema: RawActionData;
+}) => {
+  const modifiedSchema = { ...toolSchema };
+  modifiedSchema.parameters = {
+    ...modifiedSchema.parameters,
+    required: modifiedSchema.parameters?.required?.filter(
+      field => !['project_id', 'team_id'].includes(field)
+    ) || []
+  };
+
+  return modifiedSchema;
+};
+```
+</Step><Step title="Input Modifier">
+This will be used to modify the input data for the `LINEAR_CREATE_LINEAR_ISSUE` action. Here we have added the values for `project_id` and `team_id` parameters to the input data. By doing this, we can avoid specifying these values in the prompt and be sure that the agent uses the correct values. The technical term for this is **Pre-Processing**.
+```typescript TypeScript
+const preProcessor: TPreProcessor = ({ params, actionName, appName }: {
+  params: Record<string, unknown>;
+  actionName: string;
+  appName: string;
+}) => {
+  const modifiedParams = { ...params };
+
+  modifiedParams.project_id = "e708162b-9b1a-4901-ab93-0f0149f9d805";
```

Comment:
Consider extracting these IDs into constants at the top of the file or into a configuration object for better maintainability:
```typescript
const LINEAR_CONFIG = {
  PROJECT_ID: 'e708162b-9b1a-4901-ab93-0f0149f9d805',
  TEAM_ID: '249ee4cc-7bbb-4ff1-adbe-d3ef2f3df94e'
} as const;

// Then in the preprocessor:
modifiedParams.project_id = LINEAR_CONFIG.PROJECT_ID;
modifiedParams.team_id = LINEAR_CONFIG.TEAM_ID;
```